packages/editor/src/state/callbacks.ts

```diff
@@ -707,20 +714,166 @@ export const useDeleteLocalFile = () =>
         if (!shouldDelete) {
           return;
         }
-        const currentWorkspace = snapshot.getLoadable(
-          currentWorkspaceState,
-        ).contents;
-        // removes from index
-        set(localFilesState, (localFiles) => {
-          const { [id]: removedFile, ...newFiles } = localFiles;
-          return newFiles;
-        });
-        await localforage.removeItem(id);
-        if (currentWorkspace.metadata.id === id) {
-          // set rather than reset to generate new id to avoid id conflicts
-          set(currentWorkspaceState, () => defaultWorkspaceState());
-          reset(diagramState);
+
+        // Delete from cloud storage
+        if (authObject.currentUser != null) {
+          const notif = toast.loading(`Deleting ${name}...`);
+          await deleteDoc(doc(db, authObject.currentUser.uid, id))
+            .catch((error) => {
+              toast.error(`Error deleting diagram: ${name}`);
+              toast.dismiss(notif);
+              return;
+            })
+            .then(() => {
+              // If successful, remove from local state
+              const currentWorkspace = snapshot.getLoadable(
+                currentWorkspaceState,
+              ).contents;
+              set(savedFilesState, (savedFiles) => {
+                const { [id]: removedFile, ...newFiles } = savedFiles;
```

Comment:
huh didn't know about destructuring assignment and rest properties ([docs](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Destructuring_assignment#rest_property)). pretty cool!